File: reverseString/reverseString.cs
//Coding Challenge: * **Write a function that reverses a string.** 
// This is a simple but challenging challenge that tests your understanding of strings and reversing them.


//to run this file, you must build it with dotnet

//method to reverse a string in c#
using System;

class Reverse{
    public static void ReverseString(string sentence){
        char[] letters = sentence.ToCharArray();
        Array.Reverse(letters);
        string reversed = new string(letters);
        Console.WriteLine(reversed);

    }

    //without using reverse
    public static void ReverseNoReverse(string sentence){
        char[] letters = sentence.ToCharArray();
        List<char> lettersRev = new List<char>();
        foreach(char element in letters){
            lettersRev.Insert(0, element);
        }
        string result = new string(lettersRev.ToArray());
        Console.WriteLine(result);
    }

    public static void Main(string[] args){
        ReverseString("hello world");
        ReverseNoReverse("hello world");
    }
}

File: Reverse a String/reverseString.js
//Coding Challenge: * **Write a function that reverses a string.** 
// This is a simple but challenging challenge that tests your understanding of strings and reversing them.

//javascript function to reverse a string
function reverseString(string){
    reversed = string.split('').reverse().join('');
    return reversed;
}

console.log(reverseString('hello world'));


//what if you can't use the reverse method?
function revNoReverse(string){
    reversed = string.split('');
    newArray = [];
    reversed.forEach(element => {
        newArray.unshift(element);
    });
    reversed = newArray.join('');
    return reversed;
}

console.log(revNoReverse('hello world'))